aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -428,14 +541,20 @@ impl AggregatorData {
     pub fn get_aggregator(
         &mut self,
         id: AggregatorID,
+        resolver: &dyn AggregatorResolver,
         max_value: u128,
     ) -> PartialVMResult<&mut Aggregator> {
         let aggregator = self.aggregators.entry(id).or_insert(Aggregator {
-            value: 0,
-            state: AggregatorState::PositiveDelta,
+            id,
+            state: AggregatorState::Delta {
+                speculative_start_value: 0,
+                speculative_source: SpeculativeValueSource::Default,
+                delta: DeltaValue::Positive(0),
+                history: DeltaHistory::new(),
+            },
             max_value,
-            history: Some(History::new()),
         });
+        aggregator.read_last_committed_aggregator_value(resolver);
```

Comment:
given how you've implemented this, maybe we should do this call in try_add/try_sub (i.e. snapshot() call doesn't need it, and read() call can just directly do the expensive read)

Reply:
Done.